src/ethereum_spec_tools/evm_tools/t8n/t8n_types.py

```diff
@@ -417,6 +419,14 @@ def add_transaction(self, tx: Any) -> None:
         else:
             self.successful_txs.append(tx)
 
+    def add_receipt(self, tx: Any) -> None:
+        """
+        Add t8n receipt info for valid tx
+        """
+        txhash = keccak256(rlp.encode(tx))
```

Comment:
```suggestion
        if self.t8n.is_after_fork("ethereum.berlin") and not isinstance(
            tx, self.t8n.fork_types.LegacyTransaction
        ):
            tx_hash = keccak256(self.t8n.fork_types.encode_transaction(tx))
        else:
            tx_hash = keccak256(rlp.encode(tx))
```

Reply:
For non-legacy transactions, we need to convert the transaction to bytes first and then extract the hash.